src/adapters/hasura/altUser.adapter.ts

```diff
@@ -1293,15 +1295,100 @@ export class ALTHasuraUserService {
     });
   }
 
-  sendSuccessResponse(res: any, statusCode: number, message: string, data: any) {
+  sendSuccessResponse(
+    res: any,
+    statusCode: number,
+    message: string,
+    data: any
+  ) {
     return res.status(statusCode).send({
       success: true,
       status: "Authenticated",
       message,
       data,
     });
   }
+  public async deleteUser(request, userId) {
+    let client;
+    try {
+      // Verify user token and systemAdmin role
+      const userToken = request.headers.authorization?.split(" ")[1];
+      if (!userToken) {
+        return {
+          success: false,
+          message: "Authorization token is required",
+        };
+      }
 
+      // Verify systemAdmin role
+      const decodedToken: any = jwt_decode(userToken);
+      const hasSystemAdminRole =
+        decodedToken?.resource_access?.["hasura-app"]?.roles?.includes(
+          "systemAdmin"
+        );
 
+      if (!hasSystemAdminRole) {
+        return {
+          success: false,
+          message: "Only system administrators can delete users",
+          status: 403,
+        };
+      }
+      if (
+        request.headers.delete_api_secret &&
+        request.headers.delete_api_secret !== process.env.DELETE_API_SECRET
+      ) {
+        return new ErrorResponse({
+          errorCode: "403",
+          errorMessage: "Invalid Secret Key",
+        });
+      }
+      console.log("VALIDATE KEY->>>", request.headers.delete_api_secret);
+
+      // Create connection string using environment variables
+      const connectionString =
+        process.env.TELEMETRY_DB_URL ||
+        `postgres://${process.env.TELEMETRY_DB_USER}:${encodeURIComponent(
+          process.env.TELEMETRY_DB_PASSWORD
+        )}@${process.env.TELEMETRY_DB_HOST}:${process.env.TELEMETRY_DB_PORT}/${
+          process.env.TELEMETRY_DB_NAME
+        }?sslmode=disable`;
+
+      // Create a connection pool
+      const pool = new Pool({
+        connectionString,
+      });
+
+      // Connect to the database
+      let telemetryClient = await pool.connect();
+
+      const selectTelemetryQuery = `
```

Comment:
@YoginiTayade if user_id i.e actor id is not set or its null then we should return an error message. If this is passed as null it could delete the records where its not set